staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go

```diff
@@ -100,21 +100,33 @@ func validateJWTAuthenticator(authenticator api.JWTAuthenticator, fldPath *field
 func validateIssuer(issuer api.Issuer, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 
-	allErrs = append(allErrs, validateURL(issuer.URL, fldPath.Child("url"))...)
+	allErrs = append(allErrs, validateIssuerURL(issuer.URL, fldPath.Child("url"))...)
+	allErrs = append(allErrs, validateIssuerDiscoveryURL(issuer.DiscoveryURL, fldPath.Child("discoveryURL"))...)
 	allErrs = append(allErrs, validateAudiences(issuer.Audiences, issuer.AudienceMatchPolicy, fldPath.Child("audiences"), fldPath.Child("audienceMatchPolicy"))...)
 	allErrs = append(allErrs, validateCertificateAuthority(issuer.CertificateAuthority, fldPath.Child("certificateAuthority"))...)
 
 	return allErrs
 }
 
-func validateURL(issuerURL string, fldPath *field.Path) field.ErrorList {
-	var allErrs field.ErrorList
-
+func validateIssuerURL(issuerURL string, fldPath *field.Path) field.ErrorList {
 	if len(issuerURL) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath, "URL is required"))
-		return allErrs
+		return field.ErrorList{field.Required(fldPath, "URL is required")}
 	}
 
+	return validateURL(issuerURL, fldPath)
+}
+
+func validateIssuerDiscoveryURL(issuerDiscoveryURL string, fldPath *field.Path) field.ErrorList {
```

Comment:
These need to be unique right?  Also probably not the same as the configured issuer URL?

Reply:
They need to be unique but as of this PR we still only support 1 JWT authenticator. I've a PR open to support multiple https://github.com/kubernetes/kubernetes/pull/123431 and added the issuer unique check there. I was waiting to see which PR to add the uniqueness check for discovery url based on which one got merged first.